<?xml version="1.0" encoding="UTF-8"?>
<launch>

<include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find projet2022)/worlds/Projet2022.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>


 <!-- Spwan turtlebot in Gazebo -->

<param name="robot_description" command="$(find xacro)/xacro $(find projet2022)/urdf/turtlebot3.urdf.xacro " />
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0.87 -y -1.74 -z 0.0 -Y 0 -param robot_description" /> 
<!---defaut  -x 0.87 -y -1.74 -z 0.0 -Y 0-->
<!-- couloir -x 0.96 -y 1.72 -z 0.0 -Y -1.57-->
<!-- env -x -1.766 -y 0.04 -z 0.0 -Y -1.57-->


<!-- Rviz -->
<!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/> -->
<!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/> -->
<!-- node name="rviz" pkg="rviz" type="rviz" args="-d $(find projet2022)/rviz/config.rviz"/> -->




<!--node connecte-->
<node pkg="projet2022" name="connect" type="connect.py" 
output="screen" required="true">
</node>	

<!--challenge1 and 3 -->
<node pkg="projet2022" name="challenge1and3" type="challenge1and3.py" 
output="screen" required="true">
</node>
<param name="linear_scale1" type="double" value="0.26">
</param>
<param name="angular_scale1" type="double" value="0.4">
</param>

<!--challenge2 couloir -->
<node pkg="projet2022" name="challenge2" type="challenge2.py" 
output="screen" required="true">
</node>	
<param name="linear_scale2" type="double" value="0.3">
</param>
<param name="angular_scale2" type="double" value="0.5">
</param>


</launch>